render-configs.sh

```diff
@@ -1,32 +1,136 @@
 #!/bin/bash
+### How to call this script
+# To write secrets to tmp files:
+# ./render-configs.sh (defaults to dev azure, tools RBS)
 
-LOCAL_TOKEN=$(cat ~/.vault-token)
-VAULT_TOKEN=${1:-$LOCAL_TOKEN}
+# There are three optional arguments:
+# ./render-configs.sh (Azure Synapse: -a dev|integration) (RBS: -r tools|dev)  (Put string of env variables in your clipboard to copy to intellij: -i)
+# e.g.: ./render-configs.sh -a dev -r tools -i
+# This would set azure synapse to dev, RBS to tools, and put the variables in your clipboard
 
+# If you're running Azure Integration Tests you should use the following settings:
+# ./render-configs.sh -a integration -r tools
+# Then, refresh your z-shell configuration (`source ~./zshrc`) (follow getting started doc to set env variables)
+# Alternatively, if you use the -i flag, it copies the environment variables to your clipboard and you can paste them into your Intellij test profile.
+# ./render-configs.sh -a integration -r tools -i
+
+# If you want a set up locally, you can use the following settings:
+# ./render-configs.sh -a dev -r dev
+# Then, refresh your z-shell configuration (`source ~./zshrc`)
+# ./gradlew bootRun
+
+AZURE_ENV=dev
+RBS_ENV=tools
+COPY_INTELLIJ_ENV_VARS=n
+
+while getopts ":a:r:i" option; do
+  case $option in
+    a)
+      AZURE_ENV=$OPTARG
+      ;;
+    r)
+      RBS_ENV=$OPTARG
+      ;;
+    i)
+      COPY_INTELLIJ_ENV_VARS=y
+      ;;
+    *)
+      echo "Usage: $0 [-a (dev|integration)] [-r (tools|dev)] [-i]"
+      exit 1
+      ;;
+  esac
+done
+
+# ========================
+# Azure Credentials
+# ========================
+# If you want to run Azure Integration tests locally, you need to point to the "integration" environment
+# Options: 'dev', 'integration'
+
+if [[ "${AZURE_ENV}" == "dev" ]]; then
+    AZURE_SYNAPSE_WORKSPACENAME=tdr-synapse-east-us-ondemand.sql.azuresynapse.net
+elif [[ "${AZURE_ENV}" == "integration" ]]; then
+    AZURE_SYNAPSE_WORKSPACENAME=tdr-snps-int-east-us-ondemand.sql.azuresynapse.net
+else
+    echo "Invalid Azure environment: $AZURE_ENV"
+    exit 1
+fi
+# writing this values to a tmp file so the value can match the set azure environment
+echo $AZURE_SYNAPSE_WORKSPACENAME > "/tmp/azure-synapse-workspacename.txt"
+
+
+vault read -field=tenant-id secret/dsde/datarepo/"$AZURE_ENV"/azure-application-secrets \
+    > "/tmp/jade-dev-tenant-id.key"
+AZURE_CREDENTIALS_HOMETENANTID=$(cat "/tmp/jade-dev-tenant-id.key")
```

Comment:
Not sure how much tweaking you want to do with this now, but since all of these secrets are from the same vault path you can instead read them all at once and then pull out the fields you want from your local copy.

```
AZURE_SECRETS=$(vault read -field data -format json secret/dsde/datarepo/"$AZURE_ENV"/azure-application-secrets)

AZURE_CREDENTIALS_HOMETENANTID=$(echo "$AZURE_SECRETS" | jq -r '."tenant-id"')
AZURE_CREDENTIALS_APPLICATION_ID=$(echo "$AZURE_SECRETS" | jq -r '."client-id"')
...
```

Reply:
Thanks for taking a close look at this! 
It would be ideal to only need to need to pull the secret once, but I don't know that I find this echo/jq pattern much cleaner. And I still do need to write to the tmp files for our environment variables. I think I'm inclined to leave this piece as is.